x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/rerank/TextSimilarityRankBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.rerank;
+
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankBuilder;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.search.rank.context.QueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.QueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingRankFeaturePhaseRankShardContext;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public class TextSimilarityRankBuilder extends RankBuilder {
+
+    private final String inferenceId;
+    private final String inferenceText;
+    private final String field;
+    private final float minScore;
+
+    public TextSimilarityRankBuilder(String field, String inferenceId, String inferenceText, int rankWindowSize, float minScore) {
+        super(rankWindowSize);
+        this.inferenceId = inferenceId;
+        this.inferenceText = inferenceText;
+        this.field = field;
+        this.minScore = minScore;
+    }
+
+    public TextSimilarityRankBuilder(StreamInput in) throws IOException {
+        super(in);
+        this.inferenceId = in.readString();
+        this.inferenceText = in.readString();
+        this.field = in.readString();
+        this.minScore = in.readFloat();
```

Comment:
IIUC if we don't specify a `minScore` we would have a neg inf value, right? Maybe we could use `Float` instead and just have it as null and read/write `optionalFloat` instead? (+ maybe we could also return true for the predicate in this scenario w/o any comparisons)?

Reply:
Switched to optional Float.